src/mlpack/core/data/csv_parser_impl.hpp

```diff
@@ -0,0 +1,202 @@
+/* This file is originated from armadillo and adapted for mlpack
+ * https://gitlab.com/conradsnicta/armadillo-code/-/blob/10.5.x/include/armadillo_bits/diskio_meat.hpp
+ * Copyright 2008-2016 Conrad Sanderson (http://conradsanderson.id.au)
+ * Copyright 2008-2016 National ICT Australia (NICTA)
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ------------------------------------------------------------------------
+*/
+#ifndef MLPACK_CORE_DATA_CSV_PARSER_IMPL_HPP
+#define MLPACK_CORE_DATA_CSV_PARSER_IMPL_HPP
+
+#include "csv_parser.hpp"
+
+namespace mlpack
+{
+namespace data
+{
+
+  /**
+   * Given the address of a martix element(val)
+   * sets it equal to the provided value(token)
+   * example calling: convert_token<eT>(x.at(row, col), token)
+   */
+  template<typename MatType>
+  bool ConvertToken(typename MatType::elem_type& val, const std::string& token)
+  {
+    const size_t N = size_t(token.length());
+
+    if (N == 0)
+    {
+      val = typename MatType::elem_type(0);
+
+      return true;
+    }
+
+    const char* str = token.c_str();
+
+    // checking for nan, +inf, -inf
+    // in both upper and lower case
+    // using arma::Datum which basically
+    // contains all the physical constants
```

Comment:
Would you improve comments here too? Be sure to start with a capital letter and end with a comma,
Do not hesitate in being more descriptive, this will allow you and me to understand in the future why you wrote this. :+1:

Reply:
Yea, I will write detailed comments about the implementations.